use go workspace

codeboten opened the pull request:
This change removes the need for `replace` statements in all modules within this repository. Some additional changes needed to make this work:
    
- ~update example docker base image~
- ~update compliance tests to avoid issue w/ go workspace~

bogdandrutu commented:
@codeboten are we going to have this 👯 ?

codeboten commented:
@bogdandrutu yes I'll tackle this as soon as I can get https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/12918 merged

jpkrohling commented on examples/tracing/Dockerfile:
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.17 AS build
+FROM golang:1.18 AS build
```
Should we just go for 1.19 instead?

jpkrohling replied:
Answering my question: no.

jpkrohling commented on go.work:
```diff
@@ -0,0 +1,382 @@
+// 	references to modules that are in this repository with their relative paths
+// so that we always build with current (latest) version of the source code.
+
+replace (
```
I assume this would cause the release process to change a bit. If that's the case, can we have an update to it as part of this PR?

codeboten replied:
Absolutely.

mx-psi commented on go.work:
```diff
@@ -0,0 +1,382 @@
+// 	references to modules that are in this repository with their relative paths
+// so that we always build with current (latest) version of the source code.
+
+replace (
```
Do we actually need these `replace` statements? My impression from reading https://go.dev/doc/tutorial/workspaces is that the `use` statements were enough

codeboten replied:
These are no longer needed, they've been removed

Commit:
use go workspace

This change removes the need for replace statements in all modules within this repository.

Commit:
add internal/tools

Commit:
add extension/jaegerremotesampling

Commit:
update go workspace

Commit:
go work sync

codeboten commented:
Closing this for now, there's no clear path to move forward with workspaces in this repo as it adds unnecessary complexity to the release process.
